### ISSUE TITLE
Memory Leak in TFB_Canvas

### ISSUE THREAD

**Issue by Serosis** [JHGuitarFreak/UQM-MegaMod-Archived#6]
It has been brought to my attention that there's a memory leak in TFB_Canvas. It crashes the game from between 45-70 minutes of play when entering a solar system from Hyperspace or entering orbit of a planet/moon.

I have not been able to replicate this myself, as I do not play it for long enough, but I'm throwing this out there to make people aware that it exists.

**Comment by Serosis** [comment]
I found a line commented out in solarsys.c which frees up the IP data. I've uncommented it to see if this may fix the leak.
____
That did not fix it.

**Comment by Ala-lala** [comment]
After additional testing, I think this was indeed fixed by #22, unless it's very option-specific. I've tested:
* Playing for hours at a time without quitting [original graphics, PC music] (the only leak I found was the space music leak, which is now fixed and wasn't severe enough to crash the game)

(On the below playthroughs, I had already implemented #23, but that leak isn't frequent enough to be the cause of a crash)

* Playing the game from start to finish without loading or quitting it, taking about five hours of playtime (plus a few hours paused, excluding time when my computer was suspended) [HD graphics, PC music]
* Loading a late save from the above playthrough and playing until the end while reloading often (whenever I lost crew), taking about ninety minutes [HD graphics, PC? music]
* Loading an even later save and playing until the end, taking about 20-30 minutes [HD graphics, 3DO music]
* Doing the above again with [HD graphics, Precursors music + alternate hyperspace/quasispace music]

The process memory never went above ~650 MB except at the ending explosion cutscene (which was properly freed after it finished).
